bin/travis-build.bash

```diff
@@ -1,75 +1,123 @@
 #!/bin/bash
 set -e
-set -x # echo on
 
 echo "This is travis-build.bash..."
 
 echo "Installing the packages that CKAN requires..."
 sudo apt-get update -qq
-sudo apt-get install solr-jetty libcommons-fileupload-java
+sudo apt-get install -y solr-jetty libcommons-fileupload-java
 
-echo "Upgrading libmagic for ckanext-qa..."
-# appears to upgrade it from 5.09-2 to 5.09-2ubuntu0.6 which seems to help the tests
-sudo apt-get install libmagic1
+# !/bin/bash
+ver=$(python -c"import sys; print(sys.version_info.major)")
+if [ $ver -eq 2 ]; then
+    echo "python version 2"
+elif [ $ver -eq 3 ]; then
+    echo "python version 3"
+else
+    echo "Unknown python version: $ver"
+fi
 
 echo "Installing CKAN and its Python dependencies..."
-git clone https://github.com/ckan/ckan
-cd ckan
+if [ ! -d ckan ]; then
 
-if [ $CKANVERSION == 'master' ]
-then
-    echo "CKAN version: master"
-else
-    CKAN_TAG=$(git tag | grep ^ckan-$CKANVERSION | sort --version-sort | tail -n 1)
-    git checkout $CKAN_TAG
-    echo "CKAN version: ${CKAN_TAG#ckan-}"
+  echo "Creating the PostgreSQL user and database..."
+  sudo -u postgres psql -c "CREATE USER ckan_default WITH PASSWORD 'pass';"
+  sudo -u postgres psql -c 'CREATE DATABASE ckan_test WITH OWNER ckan_default;'
+
+  if [ "${CKAN_BRANCH}dd" == 'dd' ]; then
+    #remote lookup tags, and get latest by version-sort
+    CKAN_TAG=$(git ls-remote --tags https://github.com/$CKAN_GIT_REPO | grep refs/tags/ckan-$CKANVERSION | awk '{print $2}'| sort --version-sort | tail -n 1 | sed  's|refs/tags/||' )
+    echo "CKAN tag version $CKANVERSION is: ${CKAN_TAG#ckan-}"
+    cmd="git clone --depth=50 --branch=$CKAN_TAG https://github.com/$CKAN_GIT_REPO ckan"
+    echo $cmd
+    eval $cmd
+  else
+    echo "CKAN version: $CKAN_BRANCH"
+    cmd="git clone --depth=50 --branch=$CKAN_BRANCH https://github.com/$CKAN_GIT_REPO ckan"
+    echo $cmd
+    eval $cmd
```

Comment:
If we're using `eval`, then should we quote `$cmd`?

Reply:
should drop this down again, as its not required post testing